four_wheel_steering_controller/src/four_wheel_steering_controller.cpp

```diff
@@ -220,26 +220,26 @@ namespace four_wheel_steering_controller{
     controller_nh.param("angular/z/min_acceleration"       , limiter_ang_.min_acceleration       , -limiter_ang_.max_acceleration      );
 
     // If either parameter is not available, we need to look up the value in the URDF
-    bool lookup_track = !controller_nh.getParam("track", track_);
-    bool lookup_wheel_radius = !controller_nh.getParam("wheel_radius", wheel_radius_);
-    bool lookup_wheel_base = !controller_nh.getParam("wheel_base", wheel_base_);
+    bool lookup_track = !root_nh.getParam("track", track_);
+    bool lookup_wheel_radius = !root_nh.getParam("wheel_radius", wheel_radius_);
+    bool lookup_wheel_base = !root_nh.getParam("wheel_base", wheel_base_);
 
     urdf_geometry_parser::UrdfGeometryParser uvk(root_nh, base_frame_id_);
     if(lookup_track)
       if(!uvk.getDistanceBetweenJoints(front_steering_names[0], front_steering_names[1], track_))
         return false;
       else
-        controller_nh.setParam("track",track_);
+        root_nh.setParam("track",track_);
     if(lookup_wheel_radius)
       if(!uvk.getJointRadius(front_wheel_names[0], wheel_radius_))
         return false;
       else
-        controller_nh.setParam("wheel_radius",wheel_radius_);
+        root_nh.setParam("wheel_radius",wheel_radius_);
     if(lookup_wheel_base)
       if(!uvk.getDistanceBetweenJoints(front_wheel_names[0], rear_wheel_names[0], wheel_base_))
         return false;
       else
-        controller_nh.setParam("wheel_base",wheel_base_);
+        root_nh.setParam("wheel_base",wheel_base_);
```

Comment:
I would remove this commit...`root_nh` here points to global param space, right? 
Those parameters should _especially_ be in a namespaced environment

Reply:
The thing is that I want to use those parameters in other nodes (so they don't have to parse again the urdf).
When the parameters are in the global space, other nodes do not need to know the controller node name to find them...
Maybe I could use service instead.

Reply:
The param server is exactly for this kind of stuff. And there's nothing wrong with having to know the name of the controller node, that's not something you'd change anyways. I usually use `mobile_base_controller` as name for that on most robots regardless of what exact type of base controller that is.

Reply:
Note with researcher hat on: Keep in mind that services for your experiments are very bad as they are not transparent to anything. At least rosparam can be dumped to a file for processing later.